pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/tasks/action/PipWheelActionTest.groovy

```diff
@@ -0,0 +1,143 @@
+    }
+
+    def 'wheel install uses global options'() {
+        Map<String, List<String>> override = ['setuptools': ['--global-option', '--dummy-global-option']]
+        def settings = new PipActionHelpers.GlobalOptionOverridePackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("setuptools-1.0.0.tar.gz"), [])
+
+        then:
+        1 * execSpec.commandLine(_) >> { List<List<String>> args ->
+            println args
+            def it = args[0]
+            def idx = it.indexOf('--global-option')
+            assert idx != -1
+            assert it[2] == 'wheel'
+            assert it[idx + 1] == '--dummy-global-option'
+        }
+    }
+
+    def 'wheel install uses build options'() {
+        Map<String, List<String>> override = ['setuptools': ['--build-option', '--disable-something']]
+        def settings = new PipActionHelpers.BuildOptionPackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("setuptools-1.0.0.tar.gz"), [])
+
+        then:
+        1 * execSpec.commandLine(_) >> { List<List<String>> args ->
+            println args
+            def it = args[0]
+            def idx = it.indexOf('--build-option')
+            assert idx != -1
+            assert it[2] == 'wheel'
+            assert it[idx + 1] == '--disable-something'
+        }
+    }
+
+    def "wheel install uses supported language versions"() {
+        Map<String, List<String>> override = ['setuptools': ['2.8']]
+        def settings = new PipActionHelpers.SupportedLanguageVersionOverridePackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("setuptools-1.0.0.tar.gz"), [])
+
+        then:
+        def e = thrown(PipExecutionException)
+        e.message == 'Package setuptools works only with Python versions: [2.8]'
+    }
+
+    def 'wheel install requires source rebuild'() {
+        List<String> override = ['pyflakes']
+        def settings = new PipActionHelpers.RequiresRebuildOverridePackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("pyflakes-1.0.0.tar.gz"), [])
+
+        then:
+        1 * execSpec.commandLine(_) >> { List<List<String>> args ->
+            println args
+            def it = args[0]
+            def idx = it.indexOf('--no-deps')
+            assert idx != -1
+            assert it[2] == 'wheel'
+            assert !it.any { entry -> entry == '--ignore-installed' }
```

Comment:
Sure, but this is not confirming that the wheel was actually required to be rebuilt/reinstalled.
This test prompted me to put the comments above about lack of integration test quality with these tests.

Reply:
Fixed. Made the `WheelCache` a mock, and asserting there are no interactions with it.